independent-projects/arc/runtime/src/main/java/io/quarkus/arc/FixedValueSupplier.java

```diff
@@ -7,7 +7,8 @@
  *
  * @author Maarten Mulders
  */
-public class FixedValueSupplier implements Supplier {
+public class FixedValueSupplier implements Supplier<Object> {
```

Comment:
Maybe we should give this `Supplier` a more specific type parameter, too. Although it doesn't really add much value at runtime ;-)

Reply:
I'd keep this universal.

Reply:
Then maybe make this one a `FixedValueSupplier<T>`, just like the `MapBeanSupplier`. Still universal, but cleaner when it comes to using generics, I'd say?